src/HasteModuleLoader/HasteModuleLoader.js

```diff
@@ -608,7 +616,10 @@ class Loader {
       runOnlyPendingTimers: () =>
         this._environment.fakeTimers.runOnlyPendingTimers(),
 
-      setMock: (moduleName, moduleExports) => {
+      setMock: (moduleName, moduleExports, options) => {
+        if (options && options.virtual) {
+          this._virtualModules[moduleName] = moduleExports;
```

Comment:
we can't store `moduleExports` here as I explained before. This should be a path, because you are calling it in `_resolveModuleName`.

Would it be enough if we just don't throw here: https://github.com/facebook/jest/pull/740/files#diff-c57162543a4f0bc311cba66ba52baf7bR388 if `virtualModules[moduleName] === true`?